sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RequestOptions.java

```diff
@@ -595,6 +602,15 @@ public AtomicReference<Runnable> getMarkE2ETimeoutInRequestContextCallbackHook()
         return this.markE2ETimeoutInRequestContextCallbackHook;
     }
 
+    public void setKeywordIdentifiers(Set<String> keywordIdentifiers) {
+        this.keywordIdentifiers = keywordIdentifiers;
```

Comment:
You should make sure to always store an immutable Set here - via Collections.unmodifiableSet(keywordIdentifiers). You are returning the same Set from the get-methods - and if the set is mutable that creates unnecessary race conditions and confusion on thread safety - by storing an immutable snapshot in the set-method you can safely return the set in the get-method, you do not have to worry about thread-safety etc.

Reply:
It would be fine to do this (wrapping in immutable set) in the external Request Options as well - but I assume it will be easier here - given that it is a central place. Your choice.

Reply:
Changed all the public setters to assign a Collections.unmodifiableSet()